goalgorilla/open_social
# Issue #3123324 by SV: As a CM I want to filter items relevant to a specific tag on the stream block

## Problem
- Should be possible to add activities blocks that shows items filtered base on specific tags on the landing/dashboard pages

## Solution
- Create additional views display plugin with dependent taxonomy filter for activity views block  

## Issue tracker
- https://getopensocial.atlassian.net/browse/YANG-2580
- https://getopensocial.atlassian.net/browse/YANG-3463
- https://www.drupal.org/project/social/issues/3123324

## How to test
- [ ]  Click edit layout on any dashboard node
- [ ]  Scroll to any section and click to "Add block"
- [ ]  After click - dashboard panel should be displayed at the right side
- [ ]  Select at the bottom an activity stream views block (without post)
- [ ]  On the adding block form should be displayed optional dependent dropdown to filter items in the block 
   - first dropdown: select any Taxonomy Vocabulary
   - second multiple dropdown: select any Taxonomy terms (which should belong to selected 
vocabulary from 1-st dropdown)
  -  or we skip those two selections then items shouldn't be filtered
- [ ]  After click save/update button, should be displayed activity stream block which is filtered by selected terms

## Screenshots
![Edit layout for Test dashboard   Global Dev Hub](https://user-images.githubusercontent.com/25609390/78567727-c908a280-7829-11ea-83d7-0165ff8ccb34.png)

![Edit layout for Test dashboard   Global Dev Hub (1)](https://user-images.githubusercontent.com/25609390/78567748-d1f97400-7829-11ea-96f4-05bc6bde8af2.png)

![Test dashboard   Global Dev Hub](https://user-images.githubusercontent.com/25609390/78567761-d7ef5500-7829-11ea-9968-15080623cf4c.png)
![Screenshot from 2020-12-27 21-02-12](https://user-images.githubusercontent.com/25609390/103177871-e2092600-4886-11eb-9972-21be3d555ecf.png)


## Release notes
Add optional module Social Activity Filter, which allows filter content in the "Activity stream (without post)" block by specific tags.
